javalin/src/main/java/io/javalin/plugin/tracing/JaegerPlugin.kt

```diff
@@ -0,0 +1,37 @@
+package io.javalin.plugin.tracing
+
+import io.jaegertracing.Configuration
+import io.jaegertracing.internal.JaegerTracer
+import io.jaegertracing.internal.samplers.ProbabilisticSampler
+import io.javalin.Javalin
+import io.javalin.core.plugin.Plugin
+import io.opentracing.util.GlobalTracer
+
+class JaegerPlugin @JvmOverloads constructor(
+        private val tracer: JaegerTracer = Configuration("javalin-app")
+                .tracerBuilder
+                .withSampler(ProbabilisticSampler(0.8))
+                .build()
+) : Plugin {
+    override fun apply(app: Javalin) {
+        // register tracer as global tracer
+        GlobalTracer.registerIfAbsent(tracer)
+        app.before {
+            // start root span
+            val span = tracer.buildSpan("root-span")
+                    .withTag("method", it.method())
+                    .withTag("path", it.path())
+                    .start()
+            // set as active span in scope manager
+            tracer.scopeManager().activate(span)
+        }
+        // finish root span
+        app.after { tracer.scopeManager().activeSpan().finish() }
+
+        // in case of exception mark active span as error - and log the exception message
+        // note this won't get triggered if the application also defines an exception handler
+        app.exception(Exception::class.java) { e, _ ->
```

Comment:
I'm not sure about this, it seems like it could be lead to confusing behavior for users. If they have uncaught exceptions now they will just show up as empty responses.

Reply:
I can remove it

or send back a 500? - sending back a 500 will help with tracing since that would get tagged